src/utils/polygon.cpp

```diff
@@ -304,158 +304,72 @@ void PolygonRef::simplify(int smallest_line_segment_squared, int allowed_error_d
-            //Skip the line, if:
-            // - the line is too short and removing the line produces small enough error, or...
-            // - removing the line produces NO error, meaning that two line segments are exactly parallel.
-            if ((here_is_beyond_line == 0 && error2 < allowed_error_distance_squared) || error2 == 0)
-            {
-                // don't add [here] to the result
-                // skip checking whether the next point has to be removed for now
-                skip(poly_idx);
-            }
-            else
-            {
-                result_list_poly.emplace_back(here);
-                prev = result_list_poly.back();
-            }
+            next = new_path[0]; //Spill over to new polygon for checking removed area.
         }
-    }
+        else
+        {
+            break; //New polygon also doesn't have any vertices yet, meaning we've completed the loop without adding any vertices. The entire polygon is too small to be significant.
+        }
+        accumulated_area_removed += current.X * next.Y - current.Y * next.X; //Shoelace formula for area of polygon per line segment.
 
-    { // stage II: keep removing skipped verts (and skip the next vert if it was a skipped vert)
-        auto skip = [&skipped_verts, &result_list_poly](unsigned int& skipped_vert_idx, const ListPolyIt skipped_vert, std::vector<ListPolyIt>& new_skipped_verts)
+        if (length2 < smallest_line_segment_squared)
         {
-            unsigned int next_skipped_vert_idx = skipped_vert_idx + 1;
-            if (next_skipped_vert_idx < skipped_verts.size() && skipped_vert.next() == skipped_verts[next_skipped_vert_idx])
+            const coord_t area_removed_so_far = std::abs(accumulated_area_removed + next.X * previous.Y - next.Y * previous.X); //Close the polygon.
+            if (area_removed_so_far <= allowed_error_distance_squared)
             {
-                skipped_vert_idx++;
-                new_skipped_verts.emplace_back(skipped_verts[next_skipped_vert_idx]);
+                continue; //Remove the vertex.
             }
-            result_list_poly.erase(skipped_vert.it);
-        };
-        while (!skipped_verts.empty() && result_list_poly.size() >= 3)
-        {
-            std::vector<ListPolyIt> new_skipped_verts;
+        }
+        //Don't remove the vertex.
 
-            for (unsigned int skipped_vert_idx = 0; skipped_vert_idx < skipped_verts.size(); skipped_vert_idx++)
+        //But we may need to remove the previous vertex if it happens to be almost exactly in line with the new vertex.
+        if (new_path.size() >= 2)
+        {
+            const coord_t distance = LinearAlg2D::getDist2FromLineSegment(new_path[new_path.size() - 2], new_path[new_path.size() - 1], current);
+            if (distance < 25) //Less than 5 microns off, so this falls within our rounding error range.
```

Comment:
While I don't think it's serious enough to send it back for, I don't really like the hardcoded value here. Maybe it could be a constexpr?

Reply:
It turned out that this was dysfunctional when dealing with polygons that started off with 5 micron parts. I've reworked this part.